pkg/servicetest/servicetest_test.go

```diff
@@ -0,0 +1,135 @@
+package servicetest_test
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"path/filepath"
+
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/drud/ddev/pkg/dockerutil"
+	"github.com/drud/ddev/pkg/fileutil"
+	"github.com/drud/ddev/pkg/plugins/platform"
+	"github.com/drud/ddev/pkg/testcommon"
+	"github.com/drud/ddev/pkg/util"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	TestSites = []testcommon.TestSite{
+		{
+			Name:                          "TestMainPkgDrupal8",
+			SourceURL:                     "https://github.com/drud/drupal8/archive/v0.6.0.tar.gz",
+			ArchiveInternalExtractionPath: "drupal8-0.6.0/",
+		},
+	}
+	ServiceFiles []string
+	ServiceDir   string
+)
+
+func TestMain(m *testing.M) {
+	for i := range TestSites {
+		err := TestSites[i].Prepare()
+		if err != nil {
+			log.Fatalf("Prepare() failed on TestSite.Prepare(), err=%v", err)
+		}
+	}
+
+	log.Debugln("Running tests.")
+	testRun := m.Run()
+
+	for i := range TestSites {
+		TestSites[i].Cleanup()
+	}
+
+	os.Exit(testRun)
+}
+
+func TestServicesSetup(t *testing.T) {
+	if !testing.Short() {
+		var err error
+		assert := assert.New(t)
+
+		ServiceDir, err = filepath.Abs("../../services")
+		assert.NoError(err)
+
+		err = filepath.Walk(ServiceDir, func(path string, f os.FileInfo, _ error) error {
+			if !f.IsDir() && strings.HasPrefix(f.Name(), "docker-compose") {
+				ServiceFiles = append(ServiceFiles, f.Name())
+			}
+			return nil
+		})
+		assert.NoError(err)
+	} else {
+		fmt.Println("services tests skipped in short mode")
+	}
+}
+
+// TestServices tests each service compose file in the services folder.
+// It tests that a site can fully start w/ the compose file present, and
+// checks that any exposed HTTP ports return 200.
+func TestServices(t *testing.T) {
+	assert := assert.New(t)
+
+	for _, service := range ServiceFiles {
+		fmt.Println("Service tests for ", service)
+
+		serviceName := strings.TrimPrefix(service, "docker-compose.")
+		serviceName = strings.TrimSuffix(serviceName, ".yaml")
+
+		for _, site := range TestSites {
```

Comment:
Agreed that it may show potential conflicts, but that may be a plus. It feels likely that users needing additional services might be advanced users and could need more than one (i.e. both solr and redis). Identifying conflicts between services, unlikely as they may be, could be a benefit.